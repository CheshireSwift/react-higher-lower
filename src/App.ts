import { div } from "react-hyperscript-helpers";

const App = () => div("Hi!");

export default App;
