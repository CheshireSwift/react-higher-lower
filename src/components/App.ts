import { h } from "react-hyperscript-helpers";
import { createStore } from "redux";
import { Provider } from "react-redux";

import { rootReducer } from "../state/reducers";
import Layout from "./Layout";

const store = createStore(rootReducer);

const App = () => h(Provider, { store }, [h(Layout)]);

export default App;
