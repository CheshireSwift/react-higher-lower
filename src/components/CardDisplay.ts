import { div } from "react-hyperscript-helpers";

import { Card, Rank } from "../state/cards";

interface CardDisplayProps {
  card: Card | null;
  style?: any;
}

function formatRank(rank: Rank) {
  switch (rank) {
    case 11:
      return "J";
    case 12:
      return "Q";
    case 13:
      return "K";
    case 1:
      return "A";
    default:
      return rank;
  }
}

function formatCard(card: Card): string {
  return formatRank(card.rank) + card.suit;
}

export const CardDisplay = (props: CardDisplayProps) =>
  div(
    {
      style: {
        ...props.style,
        width: 40,
        height: 60,
        border: "1px solid black",
        background: props.card
          ? "white"
          : `radial-gradient(circle at 0% 50%, rgba(96, 16, 48, 0) 9px, #613 10px, rgba(96, 16, 48, 0) 11px) 0px 10px,
            radial-gradient(at 100% 100%,      rgba(96, 16, 48, 0) 9px, #613 10px, rgba(96, 16, 48, 0) 11px),
            #8a3`,
        backgroundSize: "20px 20px"
      }
    },
    props.card && formatCard(props.card)
  );

export default CardDisplay;
