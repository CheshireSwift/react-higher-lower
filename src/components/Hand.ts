import { div } from "react-hyperscript-helpers";
import { connect } from "react-redux";

import { State } from "../state";
import { Card, Rank } from "../state/cards";
import { cardStyle } from "./style";

interface HandProps {
  cards: Card[];
}

function formatRank(rank: Rank) {
  switch (rank) {
    case 11:
      return "J";
    case 12:
      return "Q";
    case 13:
      return "K";
    case 1:
      return "A";
    default:
      return rank;
  }
}

export const Hand = (props: HandProps) =>
  div(
    props.cards.map(card =>
      div(
        { style: { ...cardStyle, display: "inline-block", marginRight: 8 } },
        formatRank(card.rank) + card.suit
      )
    )
  );

export default connect((state: State) => ({ cards: state.hand }))(Hand);
