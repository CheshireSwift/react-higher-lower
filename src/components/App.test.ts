import { render, mount } from "enzyme";
import { h } from "react-hyperscript-helpers";

import App from "./App";
import { Hand } from "./Hand";
import { CardDisplay } from "./CardDisplay";

describe("the app", () => {
  it("matches the snapshot", () => {
    expect(render(h(App))).toMatchSnapshot();
  });

  it("starts with a card drawn", () => {
    const app = mount(h(App));

    expect(app.text()).toContain(51);
    expect(app.find(Hand).find(CardDisplay)).toHaveLength(1);
  });

  it("moves a card from hand to deck when draw is clicked", () => {
    const app = mount(h(App));
    app
      .findWhere(child => child.type() === "button" && child.text() === "draw")
      .simulate("click");

    expect(app.text()).toContain(50);
    expect(app.find(Hand).find(CardDisplay)).toHaveLength(2);
  });
});
