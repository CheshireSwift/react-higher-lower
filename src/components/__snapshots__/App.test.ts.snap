// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the app matches the snapshot 1`] = `
<div>
  <div>
    <div
      style="width:40px;height:60px;border:1px solid black"
    >
      52
    </div>
    <button>
      draw
    </button>
  </div>
  <div />
</div>
`;
